llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -5343,9 +5343,13 @@ def VImm8000: PatLeaf<(AArch64mvni_msl (i32 127), (i32 264))>;
 // trunc(umin(X, 255)) -> UQXTRN v8i8
 def : Pat<(v8i8 (trunc (umin (v8i16 V128:$Vn), (v8i16 VImmFF)))),
           (UQXTNv8i8 V128:$Vn)>;
+def : Pat<(v8i8 (truncusat (v8i16 V128:$Vn))),
```

Comment:
Can the pattern above be dropped?

Reply:
This is not possible with the current logic. However, I think it would be possible with additional logic. Would you like to see it included in this patch?

Reply:
What are the t18 and t21 nodes. It looks like you only pasted part of the DAG.

Reply:
@topperc That's weird, but I didn't touch anything else.

Reply:
@topperc I tested this by implementing additional code to support `fp_to_[su]int_sat`. As a result, if `LegalOperations` is unchecked, the associated Pattern can be deleted. 

Should we continue to check `LegalOperations`? I can not sure because I'm not a veteran.